### ISSUE TITLE
Deno support 

### ISSUE THREAD

**Issue by SOVLOOKUP** [malijs/mali#241]
Any possibility to support deno?

**Comment by anonrig** [comment]
Yes of course. Since I'm quite new to deno world, I can't foresee the expectations and the road map on it, but if can provide a roadmap for us, I'll be happy to work on it.

**Comment by SOVLOOKUP** [comment]
Deno’s roadmap about grpc:
 - Deno will support http2 in version 1.9
https://github.com/denoland/deno/issues/3995

 - And then they will implement grpc.
https://github.com/denoland/deno/issues/3326

This should probably be done this year.

But grpc will be the build-in module of deno which means you maybe must reimplement 
Mali for deno with esm and typescript.

I think deno is a better runtime for rpc micro services. Safe and decentralized. It’s worth.

**Comment by anonrig** [comment]
Sounds interesting. Upon investigating, I didn't see any difference in terms of benchmarks when compared to Node.js (referencing https://choubey.medium.com/performance-comparison-deno-vs-node-js-part-1-hello-world-3f3b26dd98b9)

Can you reference a benchmark or a major reason for developers to convert from Deno to Node.js? (I know this is not what you're signed up here for, but I wanted to clarify the reasoning behind adding Deno support since it requires a major refactoring, and integration)

**Comment by likern** [comment]
> Sounds interesting. Upon investigating, I didn't see any difference in terms of benchmarks when compared to Node.js (referencing https://choubey.medium.com/performance-comparison-deno-vs-node-js-part-1-hello-world-3f3b26dd98b9)
> 
> Can you reference a benchmark or a major reason for developers to convert from Node.js to Deno? (I know this is not what you're signed up here for, but I wanted to clarify the reasoning behind adding Deno support since it requires a major refactoring, and integration)

I'm totally not expert neither from Deno nor from Node. I do mostly Web and React Native.

We can talk about how cool / modern / built with Rust / performant Deno is.

But truly the most important feature is that they adhere  to modern WebAPIs. So using Deno is simpler for FullStack development.

And in terms of dev productivity it's groundbreaking. For example, they support WebGPU with the same semantics https://deno.com/blog/v1.8#experimental-support-for-the-webgpu-api. So having navigator object, etc.
